conans/assets/templates/new_v2_bazel.py

```diff
@@ -23,13 +24,18 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def validate(self):
+        if self.settings.os in ("Windows", "Macos") and self.options.shared:
+            raise ConanException("Windows and Macos needs extra BUILD configuration to be able "
+                                 "to create a shared library. Please, check this reference to "
+                                 "know more about it: https://bazel.build/reference/be/c-cpp")
+
```

Comment:
The next PR solves this issue. That one introduces changes in the template to let users create/consume shared libraries in all the platforms.